src/ajax.js

```diff
@@ -1,56 +1,55 @@
 import {parse as parseURL, format as formatURL} from './url';
 
-/**
- * Simple cross-browser ajax request function
- * https://gist.github.com/Xeoncross/7663273
+const XHR_DONE = 4;
 
- * IE 5.5+, Firefox, Opera, Chrome, Safari XHR object
+/**
+ * Simple IE9+ and cross-browser ajax request function
  *
  * @param url string url
  * @param callback object callback
  * @param data mixed data
  * @param options object
  */
 
-export const ajax = function ajax(url, callback, data, options = {}) {
-  let x;
-
-  try {
-    if (window.XMLHttpRequest) {
-      x = new window.XMLHttpRequest('MSXML2.XMLHTTP.3.0');
-    }
+export function ajax(url, callback, data, options = {}) {
 
-    if (window.ActiveXObject) {
-      x = new window.ActiveXObject('MSXML2.XMLHTTP.3.0');
-    }
+  let x,
+      method = options.method || (data ? 'POST' : 'GET'),
+      // For IE9 support use XDomainRequest instead of XMLHttpRequest.
+      useXDomainRequest = window.XDomainRequest &&
+        (!window.XMLHttpRequest || new window.XMLHttpRequest().responseType === undefined);
 
-    const method = options.method || (data ? 'POST' : 'GET');
+  if (useXDomainRequest) {
+    x = new window.XDomainRequest();
+    x.onload = handler;
+  } else {
+    x = new window.XMLHttpRequest();
+    x.onreadystatechange = handler;
+  }
 
-    if (method === 'GET' && data) {
-      let urlInfo = parseURL(url);
-      Object.assign(urlInfo.search, data);
-      url = formatURL(urlInfo);
-    }
+  if (method === 'GET' && data) {
+    let urlInfo = parseURL(url);
+    Object.assign(urlInfo.search, data);
+    url = formatURL(urlInfo);
+  }
 
-    //x = new (window.XMLHttpRequest || window.ActiveXObject)('MSXML2.XMLHTTP.3.0');
-    x.open(method, url, 1);
+  x.open(method, url);
 
+  if (!useXDomainRequest) {
```

Comment:
Might want to add a comment that says that IE 9 cross domain doesn't support cookies.